### PR TITLE
Remove text about the deprecated payment method "basic-card"

### LINKED ISSUE #9211: Remove text about the deprecated payment method "basic-card"
## What page(s) did you find the problem on?

files/en-us/_wikihistory.json
files/en-us/mozilla/firefox/experimental_features/index.html
files/en-us/web/api/addresserrors/index.md
files/en-us/web/api/basiccardrequest/*
files/en-us/web/api/basiccardresponse/*
files/en-us/web/api/payment_request_api/concepts/index.md
files/en-us/web/api/payment_request_api/index.md
files/en-us/web/api/payment_request_api/using_the_payment_request_api/index.md
files/en-us/web/api/paymentaddress/index.md
files/en-us/web/api/paymentmethodchangeevent/index.md
files/en-us/web/api/paymentmethodchangeevent/paymentmethodchangeevent/index.md
files/en-us/web/api/paymentrequest/canmakepayment/index.md
files/en-us/web/api/paymentrequest/paymentrequest/index.md
files/en-us/web/api/paymentresponse/details/index.md
files/en-us/web/api/paymentresponse/index.md
files/en-us/web/api/paymentvalidationerrors/index.md
files/en-us/web/security/secure_contexts/features_restricted_to_secure_contexts/index.html
files/jsondata/GroupData.json
files/jsondata/SpecData.json

## What is the problem?

"basic-card" has been deprecated from the w3c spec[1], but MDN still has referenced to it.

[1] https://www.w3.org/TR/payment-method-basic-card/ 

## What did you expect to see?

MDN articles should not reference to basic-card any more.

### PR DESCRIPTION
**Pull request by maxlgu**
> Issue number that this PR fixes (if any). For example: 'Fixes #987654321'

Fixes #9211

> What was wrong/why is this fix needed? (quick summary only)

W3C has deprecated the "basic-card" payment method[1], so MDN should remove the corresponding text as well.

[1] https://www.w3.org/TR/payment-method-basic-card/

> Anything else that could help us review it

### CODE REVIEW

**Comment by maxlgu** [comment]
@rsolomakhin

**Comment by maxlgu** [comment]
@rsolomakhin, @jpmedley , PTAL. Thanks!

File: files/en-us/web/api/payment_request_api/index.md [hunk 1]
```diff
@@ -4,7 +4,6 @@ slug: Web/API/Payment_Request_API
 tags:
   - API
   - Commerce
-  - Credit Card
```
Comment (rsolomakhin) [inline files/en-us/web/api/payment_request_api/index.md:7]: Remaining payment methods still use credit cards, so it should be OK to leave this line as-is.

**Review by rsolomakhin — CHANGES_REQUESTED** [review]
Should some page link to Google Pay, Samsung Pay, and Apple Pay tutorials or API reference?

File: files/en-us/_wikihistory.json [hunk 2]
```diff
@@ -29306,92 +29306,6 @@
       "ayqy"
     ]
   },
-  "Web/API/BasicCardRequest": {
```
Comment (jpmedley) [inline files/en-us/_wikihistory.json:29309]: @wbamberg Is this a file contributors should be touching?

File: files/en-us/web/api/payment_request_api/index.md [hunk 3]
```diff
@@ -21,15 +21,7 @@ The Payment Request API provides a consistent user experience for both merchants
...
-- **Credentials management**: Users can manage their credit cards and shipping addresses directly in the browser. A browser can also sync these "credentials" across devices, making it easy for users to jump from desktop to mobile and back again when buying things.
-- **Consistent error handling:** The browser can check the validity of card numbers, and can tell the user if a card has expired (or is about to expire). The browser can automatically suggest which card to use based on past usage patterns or restrictions from the merchant (e.g, "we only accept Visa or Mastercard"), or allow the user to say which is their default/favorite card.
+Many problems related to online shopping-cart abandonment can be traced to checkout forms, which can be difficult and time consuming to fill out and often require multiple steps to complete. The **Payment Request API** is meant to reduce the number of steps needed to complete a payment online, potentially doing away with checkout forms. It aims to make the checkout process easier, by having payment apps remembering a user's details, which are then passed along to a merchant hopefully without requiring a HTML form.
```
Comment (jpmedley) [inline files/en-us/web/api/payment_request_api/index.md:24]:
```suggestion
Many problems related to online shopping-cart abandonment can be traced to checkout forms, which can be difficult and time consuming to fill out and often require multiple steps to complete. The **Payment Request API** is meant to reduce the number of steps needed to complete a payment online, potentially doing away with checkout forms. It aims to make the checkout process easier, by having payment apps store a user's details, which are then passed along to a merchant, hopefully without requiring an HTML form.
```

File: files/en-us/web/api/paymentaddress/index.md [hunk 4]
```diff
@@ -102,21 +102,7 @@ async function doPaymentRequest() {
...
-  "cardSecurityCode" : "999",
-  "expiryMonth" : "07",
-  "expiryYear" : "2021",
-  "billingAddress" : {
-    "country" : "GB",
-    // etc. billing address is a PaymentAddress object
-  }
-}
-```
+Once the payment flow has been triggered using {{domxref("PaymentRequest.show()")}} and the promise resolves successfully, the {{domxref("PaymentResponse")}} object available inside the fulfilled promise (`instrumentResponse` above) will have a {{domxref("PaymentResponse.details")}} property that will contain response details. This has to conform to the structure defined by the payment method provider.
```
Comment (jpmedley) [inline files/en-us/web/api/paymentaddress/index.md:105]:
```suggestion
Once the payment flow has been triggered using {{domxref("PaymentRequest.show()")}} and the promise resolves successfully, the {{domxref("PaymentResponse")}} object available from the fulfilled promise (`instrumentResponse` above) will have a {{domxref("PaymentResponse.details")}} property containing response details. This has to conform to the structure defined by the payment method provider.
```

File: files/en-us/web/api/paymentrequest/paymentrequest/index.md [hunk 5]
```diff
@@ -37,14 +37,13 @@ var paymentRequest = new PaymentRequest(methodData, details, [options]);
...
+        changes with the `supportedMethods`. For example, the Example Pay payment method
+        is selected by specifying the string `https://example.com/pay` here.
     - `data`
       - : A JSON-serializable object that provides optional information that might be
         needed by the supported payment methods. This has to conform to the type expected
-        by the payment handler indicated by `supportedMethods`. For basic
-        credit card services, this structure should match the
-        {{domxref("BasicCardRequest")}} dictionary.
+        by the payment handler indicated by `supportedMethods`. Developers need to consult
+        whoever controls the payment methods for the expected shape of the data object.
```
Comment (jpmedley) [inline files/en-us/web/api/paymentrequest/paymentrequest/index.md:46]:
```suggestion
        whomever controls the payment methods for the expected shape of the data object.
```

File: files/en-us/web/api/paymentresponse/details/index.md [hunk 6]
```diff
@@ -20,8 +20,7 @@ The **`details`** read-only property of the
 provides a payment method specific message used by the merchant to process the
 transaction and determine a successful funds transfer.
 
-This data is returned by the payment app that satisfies the payment request, and must
-conform to the structure defined in the {{domxref("BasicCardResponse")}} dictionary.
+This data is returned by the payment app that satisfies the payment request. Developers need to consult whoever controls the URL for the expected shape of the details object.
```
Comment (jpmedley) [inline files/en-us/web/api/paymentresponse/details/index.md:23]:
```suggestion
This data is returned by the payment app that satisfies the payment request. Developers need to consult whomever controls the URL for the expected shape of the details object.
```

File: files/en-us/web/api/paymentresponse/index.md [hunk 7]
```diff
@@ -18,7 +18,7 @@ The **`PaymentResponse`** interface of the [Payment Request API](/en-US/docs/We
 ## Properties
 
 - {{domxref('PaymentResponse.details')}} {{readonlyinline}} {{securecontext_inline}}
-  - : Returns a JSON-serializable object that provides a payment method specific message used by the merchant to process the transaction and determine successful fund transfer. The contents of the object depend on the payment method being used; for example, if the Basic Card payment method is used, this object must conform to the structure defined in the {{domxref("BasicCardResponse")}} dictionary.
+  - : Returns a JSON-serializable object that provides a payment method specific message used by the merchant to process the transaction and determine successful fund transfer. The contents of the object depend on the payment method being used. Developers need to consult whoever controls the URL for the expected shape of the details object.
```
Comment (jpmedley) [inline files/en-us/web/api/paymentresponse/index.md:21]:
```suggestion
  - : Returns a JSON-serializable object that provides a payment method specific message used by the merchant to process the transaction and determine successful fund transfer. The contents of the object depend on the payment method being used. Developers need to consult whomever controls the URL for the expected shape of the details object.
```

**Review by jpmedley — CHANGES_REQUESTED** [review]

**Review by rsolomakhin — APPROVED** [review]

File: files/en-us/web/api/payment_request_api/index.md (hunk 1 above)
Comment (maxlgu) [reply files/en-us/web/api/payment_request_api/index.md]: Done.

**Comment by maxlgu** [comment]
@rsolomakhin , Google Pay, Apple Pay have already lived in the payment_request_api/index.md page. I have just added Samsung Pay to the list.

**Review by jpmedley — APPROVED** [review]

File: files/en-us/_wikihistory.json (hunk 2 above)
Comment (jpmedley) [reply files/en-us/_wikihistory.json:29309]: @wbamberg I'm asking because I don't think I've ever seen a content PR that touched this file.

**Comment by maxlgu** [comment]
@jpmedley , do you know whether the reply by github-actions has anything that needs our attention? I am not familiar with submitting PR to MDN. Please let me know if any action from me is needed. Thanks!

This thread is waiting for @wbamberg 's reply. After that, an owner of this repo can merge the pull requests.

File: files/en-us/_wikihistory.json (hunk 2 above)
Comment (wbamberg) [reply files/en-us/_wikihistory.json:29309]: This file should not be edited directly but it gets edited by the [`yarn content delete`](https://github.com/mdn/content#deleting-a-document) command, which also takes care of other housekeeping like removing redirects.

Comment (wbamberg) [reply files/en-us/_wikihistory.json:29309]: It's totally fine for contributors to run `yarn content delete`.

Comment (maxlgu) [reply files/en-us/_wikihistory.json:29309]: Thanks @wbamberg.

Calling `yarn content delete Web/API/BasicCardResponse -r` and `yarn content delete Web/API/BasicCardRequest -r` leads to the same changes we have had.

Resolving this thread.

**Comment by maxlgu** [comment]
Addressed @jpmedley's concern on wiki history based on @wbamberg 's advice. So the thread is ready to review. PTAL. Thanks!

Specifically, I think the reviewer need to resolve `3 workflows awaiting approval` and merge this pull request.

**Pull request merged**